src/main/java/com/xiaomi/infra/pegasus/rpc/interceptor/CompressInterceptor.java

```diff
@@ -0,0 +1,118 @@
+package com.xiaomi.infra.pegasus.rpc.interceptor;
+
+import com.xiaomi.infra.pegasus.apps.key_value;
+import com.xiaomi.infra.pegasus.apps.mutate;
+import com.xiaomi.infra.pegasus.base.error_code.error_types;
+import com.xiaomi.infra.pegasus.client.PException;
+import com.xiaomi.infra.pegasus.operator.rrdb_check_and_mutate_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_check_and_set_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_get_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_multi_get_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_multi_put_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_put_operator;
+import com.xiaomi.infra.pegasus.operator.rrdb_scan_operator;
+import com.xiaomi.infra.pegasus.rpc.async.ClientRequestRound;
+import com.xiaomi.infra.pegasus.rpc.async.TableHandler;
+import com.xiaomi.infra.pegasus.tools.ZstdWrapper;
+import java.util.List;
+
+public class CompressInterceptor implements TableInterceptor {
+  private boolean isOpen;
+
+  public CompressInterceptor(boolean isOpen) {
+    this.isOpen = isOpen;
+  }
+
+  @Override
+  public void before(ClientRequestRound clientRequestRound, TableHandler tableHandler)
+      throws PException {
+    if (!isOpen) {
+      return;
+    }
+    tryCompress(clientRequestRound);
+  }
+
+  @Override
+  public void after(
+      ClientRequestRound clientRequestRound, error_types errno, TableHandler tableHandler)
+      throws PException {
+    if (errno != error_types.ERR_OK || !isOpen) {
+      return;
+    }
+    tryDecompress(clientRequestRound);
+  }
+
+  private void tryCompress(ClientRequestRound clientRequestRound) throws PException {
+    String operatorName = clientRequestRound.getOperator().name();
+    switch (operatorName) {
+      case "put":
```

Comment:
Internally this contains a string comparison. It's inefficient. I would prefer to use type deduction:

```java
client_operator op = client.clientRequestRound.getOperator();
if(op isinstanceof rrdb_put_operator) {
}
else if(op isinstanceof rrdb_multi_put_operator) {
} else {
  throw new PException("unsupported operator = " + operatorName);
}
```